src/core/dataframe.js

```diff
@@ -9,11 +9,10 @@ export default class Dataframe {
         this.geom = geom;
         this.properties = properties;
         this.scale = scale;
-        this.size = size;
         this.type = type;
         this.decodedGeom = decoder.decodeGeom(this.type, this.geom);
-        this.numVertex = this.decodedGeom.vertices.length / 2;
-        this.numFeatures = this.decodedGeom.breakpoints.length || this.numVertex;
+        this.numVertex = type === 'point' ? size : this.decodedGeom.vertices.length / 2;
```

Comment:
Can't we just rely on `decodedGeom`? I think we were skipping the "clipped" points outside that array too.

Reply:
Can that happen with `points` type?

Reply:
Ups, forget about it: https://github.com/CartoDB/carto-vl/pull/541/files#diff-eef002f2ff8de5af0ca03149dcd70afbR477

Reply:
OK, so it seems we better keep this as is and pass the actual (used up) size of the array in `size`.